presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateCountDistinctAggregation.java

```diff
@@ -173,4 +223,21 @@ public static void evaluateFinal(@AggregationState HyperLogLogState state, Block
             BIGINT.writeLong(out, hyperLogLog.cardinality());
         }
     }
+
+    @OutputFunction(StandardTypes.BIGINT)
+    public static void evaluateFinal(BooleanDistinctState state, BlockBuilder out)
+    {
+        switch (state.getInt()) {
+            case 0:
+                BIGINT.writeLong(out, 0);
+                break;
+            case 1:
+            case 2:
+                BIGINT.writeLong(out, 1);
+                break;
+            case 3:
+                BIGINT.writeLong(out, 2);
+                break;
+        }
```

Comment:
`BIGINT.writeLong(out, Integer.bitCount( state.getInt() ))`

or handle `default` case somehow

Reply:
changed to above mentioned lines